include/fmt/core.h

```diff
@@ -32,10 +32,15 @@
 #  define FMT_GCC_PRAGMA(arg)
 #endif
 
-#if defined(__INTEL_COMPILER)
+#ifdef __ICL
+#  define FMT_ICC_VERSION __ICL
+#  define FMT_ICC_ON_WINDOWS 1
```

Comment:
I suggest replacing `FMT_ICC_ON_WINDOWS` with `FMT_ICC_POSIX` defined to 1 on non-Windows platforms. This will simplify the checks.

Reply:
Fine with me.